src/transactions/TransactionFrame.cpp

```diff
@@ -473,15 +473,29 @@ TransactionFrame::processSignatures(ValidationType cv,
 }
 
 bool
-TransactionFrame::isBadSeq(int64_t seqNum) const
+TransactionFrame::isBadSeq(LedgerTxnHeader const& header,
+                           LedgerTxnEntry const& sourceAccount, int64_t seqNum,
+                           bool applying) const
 {
+    bool hasPrev = seqNum != 0;
+    if (seqNum == 0)
+    {
+        seqNum = sourceAccount.current().data.account().seqNum;
+    }
+
+    uint32_t nextLedgerSeq = header.current().ledgerSeq + 1;
+    if (!applying && hasPrev &&
```

Comment:
The condition on `hasPrev` is annoying, this adds a new semantic meaning to `seqNum`: that `seqNum == 0` iif this is the first transaction in a sequence.